src/services/book-service.ts

```diff
@@ -0,0 +1,115 @@
+import { Request } from 'express';
+import { Document, Query, Schema, SortOrder } from 'mongoose';
+
+import { Book, IBook } from '../models/book';
+import { IBookQuery } from '../interfaces/book-query';
+import { UserRoleEnum } from '../utils/common/enum';
+import {
+  COMMON_QUERIES,
+  BOOK_NON_STRING_QUERIES
+} from '../utils/common/book-filters';
+
+import { toLowerCaseQuery } from '../utils/functions/lower-case-query';
+
+class BookService {
+  public async editSpecificBook(
```

Comment:
Hạn chế sử dụng nhiều từ đồng nghĩa

```suggestion
  public async update(
```

- Edit, Update: chỉ nên dùng update
- Find, Get, Query, Fetch: chỉ nên dùng find

Tên vừa đủ nghĩa. Ví dụ `updateBook` đặt ở đâu đó thì không sao nhưng nếu nó nằm trong 1 class liên quan trực tiếp đến Book như BookService, BookController, BookRepository thì việc nhắc đên book nó trở nên thừa thãi. Ngoài ra, ví dụ trong BookService thì mình sẽ không có các hàm kiểu `editUser`, do đó nó sẽ không bị nhập nhằng.